### PR TITLE
Fixes Ian wearing deathsquad suit

### LINKED ISSUE #9695: You are unable to put a deathsquad helmet on Ian.
Since Dsquad suits were changed to hardsuits, attempting to spawn a helmet leads to Ian's name changing, but the helmet stays stuck to your hand. Placing the hardsuit on him and using the toggle helmet button does not put a helmet on him.

### PR DESCRIPTION
**Pull request by AnturK**
Fixes #9695

### CODE REVIEW

**Comment by KazeEspada** [comment]
This means that no matter what hat, if hes dressed in deathsquid then HE IS DEATHIAN.

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 1]
```diff
@@ -271,115 +266,139 @@
...
+	if(inventory_back)
+		special_back = 1
+		switch(inventory_back.type)
+			if(/obj/item/clothing/suit/space/hardsuit/deathsquad)
+				name = "Trooper [real_name]"
+				desc = "That's not red paint. That's real corgi blood."
+			else
+				special_back = 0
+
+	if(!special_hat && !special_back)
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: these vars are only set when there's hats and a back on the corgi...
so the hat and back references could be used instead...

Comment (KazeEspada) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: No because if hes wearing something boring, then it shouldnt change anything.

Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: @KazeEspada except kaze it triggers if they have ANY HAT or ANY BACK, so they're all special, and thus the extra vars are unneeded.

Comment (AnturK) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: Scarves/Standard Armor don't cause any name changes.

Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: Oh ok, it's just that you've structured this code in such a manner that any missing cases (which you can't possible know without reading that list) are hidden in that fact that they disable the var.

brilliant.

**Comment by tedward1337** [comment]
Fun police

**Pull request merged**